ironfish-zkp/src/circuits/output.rs

```diff
@@ -366,4 +454,92 @@ mod test {
             }
         }
     }
+
+    #[test]
+    fn test_output_read_write() {
+        let mut rng = StdRng::seed_from_u64(0);
+
+        for _ in 0..5 {
+            let mut asset_id = [0u8; 32];
+            let asset_generator = loop {
+                rng.fill(&mut asset_id[..]);
+
+                if let Some(point) = asset_hash_to_point(&asset_id) {
+                    break point;
+                }
+            };
+
+            let value_commitment_randomness = jubjub::Fr::random(&mut rng);
+            let note_commitment_randomness = jubjub::Fr::random(&mut rng);
+            let value_commitment = ValueCommitment {
+                value: rng.next_u64(),
+                randomness: value_commitment_randomness,
+                asset_generator,
+            };
+
+            let nsk = jubjub::Fr::random(&mut rng);
+            let ak = jubjub::SubgroupPoint::random(&mut rng);
+            let esk = jubjub::Fr::random(&mut rng);
+            let ar = jubjub::Fr::random(&mut rng);
+
+            let proof_generation_key = ProofGenerationKey::new(ak, nsk);
+
+            let viewing_key = proof_generation_key.to_viewing_key();
+
+            let payment_address = *PUBLIC_KEY_GENERATOR * viewing_key.ivk().0;
+
+            let output = Output {
+                value_commitment: Some(value_commitment.clone()),
+                payment_address: Some(payment_address),
+                commitment_randomness: Some(note_commitment_randomness),
+                esk: Some(esk),
+                asset_id,
+                proof_generation_key: Some(proof_generation_key.clone()),
+                ar: Some(ar),
+            };
+
+            // Ser/de
+            let mut writer = vec![];
+            output.write(&mut writer).unwrap();
+            let deserialized_output: Output = Output::read(&writer[..]).unwrap();
+
+            assert_eq!(
+                output.value_commitment.is_some(),
+                deserialized_output.value_commitment.is_some()
+            );
+            if let Some(ref value_commitment) = output.value_commitment {
```

Comment:
Same comment as above: instead of checking `is_some()` and then using an `if` (whose condition will always be true), use `unwrap()` directly.